test/services/tempMailApi.test.js

```diff
@@ -61,5 +135,55 @@ describe('TempMail', () => {
         password: '1111',
       });
     });
+
+
+    it('should throw err if cant retrieve email inbox if mail and password are incorrect', async () => {
+      axios.post.mockRejectedValueOnce(new Error());
+
+
+      try {
+        await tempMailTest.getMailInbox({mail: 'enElSiguientePr@patata.net', password: '1111'});
+      } catch (error) {
+        expect(axios.post).toHaveBeenCalledTimes(1);
+        expect(axios.get).toHaveBeenCalledTimes(0);
+        expect(axios.post).toHaveBeenCalledWith('https://api.mail.tm/token', {
+          address: 'enElSiguientePr@patata.net',
+          password: '1111',
+        });
+        expect(error).toBeInstanceOf(InvalidMailInboxRetreiveError);
+        expect(error.message).toEqual('Cant get the inbox.');
+      }
+    });
+
+    it('should throw err if token is not valid', async () => {
+      axios.post
+          .mockReturnValueOnce({
+            data: {
+              token: 'GOAL',
+            },
+          });
+
+      axios.get.mockRejectedValueOnce(new Error());
```

Comment:
You're not rejecting the token here... you're making the fail on getting the inbox

Reply:
Yeah, i had a mistake, it had escaped to me